src/engine/input/key-mapping/InputGroupList.cs

```diff
@@ -0,0 +1,180 @@
+                    .ToList());
+    }
+
+    public Dictionary<string, List<InputEventWithModifiers>> GetCurrentlyPendingControlScheme()
+    {
+        return AllGroupItems.SelectMany(p => p.Actions)
+            .ToDictionary(p => p.InputName, p => p.Inputs.Select(x => x.AssociatedEvent).ToList());
+    }
+
+    public override void _Ready()
+    {
+        conflictDialog = GetNode<ConfirmationDialog>(ConflictDialogPath);
+    }
+
+    public void ShowDialog(InputEventItem caller, InputEventItem conflict, InputEventWithModifiers newEvent)
+    {
+        latestDialogCaller = caller;
+        latestDialogConflict = conflict;
+        latestDialogNewEvent = newEvent;
+
+        conflictDialog.DialogText = $"There is a conflict with {conflict.AssociatedAction.DisplayName}.\n" +
+            $"Do you want to remove the input from {conflict.AssociatedAction.DisplayName}?";
+        conflictDialog.PopupCenteredMinsize();
+    }
+
+    /// <summary>
+    ///   Get the input conflict if there are any
+    /// </summary>
+    /// <param name="item">The event with the new value</param>
+    /// <returns>The collision if any</returns>
+    public InputEventItem Conflicts(InputEventItem item)
+    {
+        // Get all environments the item is associated with.
+        var environments = item.AssociatedAction.AssociatedGroup.Environments;
+
+        // Take all InputGroups.
+        // Take the ones with any interception of the environments.
+        // Take the input actions.
+        // Get the first action where the event collides or null if there aren't any.
+        return AllGroupItems.Where(p => p.Environments.Any(x => environments.Contains(x)))
+            .SelectMany(p => p.Actions)
+            .Where(p => !Equals(item.AssociatedAction, p))
+            .SelectMany(p => p.Inputs)
+            .FirstOrDefault(p => Equals(p, item));
+    }
+
+    public void InitFromData(Dictionary<string, List<InputEventWithModifiers>> data)
+    {
+        foreach (Node child in GetChildren())
+        {
+            RemoveChild(child);
+        }
+
+        var inputGroupItemScene = GD.Load<PackedScene>("res://src/engine/input/key-mapping/InputGroupItem.tscn");
+        var inputActionItemScene = GD.Load<PackedScene>("res://src/engine/input/key-mapping/InputActionItem.tscn");
+
+        // Load json
+        var definition =
+            new[]
+            {
+                new
+                {
+                    EnvId = new[]
+                    {
+                        string.Empty,
+                    },
+                    GroupName = string.Empty,
+                    Actions = new[]
+                    {
+                        new
+                        {
+                            InputName = string.Empty,
+                            DisplayName = string.Empty,
+                        },
+                    },
+                },
+            };
+
+        using var file = new File();
```

Comment:
How does this using statement not have any parenthesis or braces?

Reply:
its an C# 8 thing. The using without braces means that the files gets closed when the current scope (the method) get left

Reply:
How does it work when we have C# target version set to 7.3?
I'm not sure if it is safe to accept this code yet? Or maybe that one guy just had outdated things? You develop on Windows, right?

Reply:
both windows and linux

Reply:
I suppose it is fine, then. Could you edit the csproj file to specify C# 8.0, then? If someone has build issues I can always then direct them to updating their C# build tools.

Reply:
Actually never mind, looks like the lang version is already set to 8.0, so everything's fine. No need for you to do anything.
I'll make a note that maybe we should port other code over to also use the C# 8 using syntax.